support/controlplane-component/controlplane-component.go

```diff
@@ -299,9 +299,18 @@ func (c *controlPlaneWorkload) defaultOptions(cpContext ControlPlaneContext, pod
 		SetDefaultSecurityContext: cpContext.SetDefaultSecurityContext,
 		Resources:                 existingResources,
 	}
-	deploymentConfig.Scheduling.PriorityClass = config.DefaultPriorityClass
-	if cpContext.HCP.Annotations[hyperv1.ControlPlanePriorityClass] != "" {
-		deploymentConfig.Scheduling.PriorityClass = cpContext.HCP.Annotations[hyperv1.ControlPlanePriorityClass]
+
+	// TODO: should all request serving omponents get APICriticalPriorityClass?
+	if c.IsRequestServing() {
+		deploymentConfig.Scheduling.PriorityClass = config.APICriticalPriorityClass
+		if priorityClass := cpContext.HCP.Annotations[hyperv1.APICriticalPriorityClass]; priorityClass != "" {
+			deploymentConfig.Scheduling.PriorityClass = priorityClass
+		}
```

Comment:
@enxebre @csrwng question for you :)

Reply:
I think it's just for the components that are in the API request path, and are required for auth (as in the oauth server). Even non-request serving components like openshift-apiserver and openshift-oauth-apiserver I would consider to be APICriticalPriority. However, something like ignition server that is request serving, I would not classify as APICriticalPriority, given that workers can just keep retrying to hit it if it's unavailable.

Reply:
I see, thanks!